vllm/v1/engine/core.py

```diff
@@ -240,17 +221,28 @@ def make_engine_core_process(
     def run_engine_core(*args, **kwargs):
         """Launch EngineCore busy loop in background process."""
 
+        def signal_handler(signum, frame):
+            raise SystemExit()
+
+        # Either SIGTERM or SIGINT will terminate the engine_core
+        signal.signal(signal.SIGTERM, signal_handler)
+        signal.signal(signal.SIGINT, signal_handler)
+
         try:
             engine_core = EngineCoreProc(*args, **kwargs)
             engine_core.run_busy_loop()
 
-        except KeyboardInterrupt:
+        except SystemExit:
             logger.debug("EngineCore interrupted.")
 
         except BaseException as e:
             logger.exception(e)
             raise e
 
+        finally:
```

Comment:
tiny nit
- If `EngineCoreProc()` fails, then `engine_core` will not exist in `finally`
- I think you can get around this by setting `engine_core = None` outside the try block and then checking if it is None in the finally block before calling shutdown